scripts/0-preinstall.sh

```diff
@@ -69,32 +69,23 @@ echo -ne "
                     Creating Filesystems
 -------------------------------------------------------------------------
 "
-createsubvolumes () {
-    btrfs subvolume create /mnt/@
-    btrfs subvolume create /mnt/@home
-    btrfs subvolume create /mnt/@var
-    btrfs subvolume create /mnt/@tmp
-    btrfs subvolume create /mnt/@.snapshots
-}
+do_btrfs() {
+    mkfs.btrfs -L "$1" "$2" -f
+    mount -t btrfs "$2" "$MOUNTPOINT"
 
-mountallsubvol () {
-    mount -o ${MOUNT_OPTIONS},subvol=@home ${partition3} /mnt/home
-    mount -o ${MOUNT_OPTIONS},subvol=@tmp ${partition3} /mnt/tmp
-    mount -o ${MOUNT_OPTIONS},subvol=@var ${partition3} /mnt/var
-    mount -o ${MOUNT_OPTIONS},subvol=@.snapshots ${partition3} /mnt/.snapshots
-}
+    echo "Creating subvolumes and directories"
+    for x in "${SUBVOLUMES[@]}"; do
+        btrfs subvolume create "$MOUNTPOINT"/"${x}" >/dev/null 2>&1
+    done
+
+    umount "$MOUNTPOINT"
+    mount -o "$MOUNT_OPTIONS",subvol=@ "$2" "$MOUNTPOINT"
 
-subvolumesetup () {
```

Comment:
Function removed but still in use by luks condition

Reply:
Thank you for letting me know.
removed it